MekHQ/src/mekhq/campaign/mission/resupplyAndCaches/Resupply.java

```diff
@@ -0,0 +1,2371 @@
+        dialog.setVisible(true);
+    }
+
+    /**
+     * Creates a start message for a contract based on its type and resupply details.
+     * The message is formatted using a specific template according to the type of
+     * the contract and whether it involves guerrilla warfare or independent command rights.
+     *
+     * @param campaign          The current campaign.
+     * @param contract          The contract for which the start message is created.
+     * @return A formatted start message for the contract, enclosed within an HTML div with a defined width.
+     */
+    private static String getContractStartMessage(Campaign campaign, AtBContract contract) {
+        int convoyCount = 0;
+        double totalCargoCapacity = 0;
+
+        for (Force force : campaign.getAllForces()) {
+            if (force.isConvoyForce() && force.isStrategicFormation()) {
+                boolean hasCargo = false;
+                for (UUID unitId : force.getAllUnits(false)) {
+                    Unit unit = campaign.getUnit(unitId);
+
+                    if (unit != null) {
+                        if (unit.isDamaged()) {
+                            continue;
+                        }
+
+                        if (!unit.isFullyCrewed()) {
+                            continue;
+                        }
+
+                        double individualCargo = unit.getCargoCapacity();
+
+                        if (!hasCargo && individualCargo > 0) {
+                            hasCargo = true;
+                        }
+
+                        totalCargoCapacity += individualCargo;
+                    }
+                }
+
+                if (hasCargo) {
+                    convoyCount++;
+                }
+            }
+        }
+
+        String convoyMessage;
+        String commanderTitle = getCommanderTitle(campaign, false);
+
+        if (contract.getContractType().isGuerrillaWarfare()) {
+            String convoyMessageTemplate = resources.getString("contractStartMessageGuerrilla.text");
+            convoyMessage = String.format(convoyMessageTemplate, commanderTitle);
+        } else {
+            String convoyMessageTemplate = resources.getString("contractStartMessageGeneric.text");
+            if (contract.getCommandRights().isIndependent()) {
+                convoyMessageTemplate = resources.getString("contractStartMessageIndependent.text");
+            }
+
+            convoyMessage = String.format(convoyMessageTemplate, commanderTitle,
+                getEstimatedCargoRequirements(campaign, contract), totalCargoCapacity, convoyCount,
+                convoyCount != 1 ? "s" : "");
+        }
+
+        int width = UIUtil.scaleForGUI(500);
+        return String.format("<html><i><div style='width: %s; text-align:center;'>%s</div></i></html>",
+            width, convoyMessage);
+    }
+}
+
+/**
+ * Class representing details about a part, including the part itself and its drop weight.
+ */
+class PartDetails {
```

Comment:
Maybe this could be a record... except by the setWeight, but we could create new PartDetails from old partDetails to have new part weights.